heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java

```diff
@@ -16,16 +16,24 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Set;
+
+import com.google.common.base.Optional;
 
 public class PackingPlan {
   private final String id;
-  private final Map<String, ContainerPlan> containers;
+  private final Map<String, ContainerPlan> containersMap;
+  private final Set<ContainerPlan> containers;
   private final Resource resource;
 
-  public PackingPlan(String id, Map<String, ContainerPlan> containers, Resource resource) {
+  public PackingPlan(String id, Set<ContainerPlan> containers, Resource resource) {
     this.id = id;
     this.containers = containers;
     this.resource = resource;
+    containersMap = new HashMap<>();
```

Comment:
Yeah, I was thinking about that. Updated to copy into an ImmutableSet during initialization.